core/lib/zksync_core/src/api_server/web3/tests/mod.rs

```diff
@@ -458,11 +458,10 @@ impl HttpTest for LogFilterChangesWithBlockBoundaries {
         };
         assert_logs_match(&lower_bound_logs, &new_events);
 
-        // FIXME(#546): update test after behavior is fixed
         let new_upper_bound_logs = client.get_filter_changes(upper_bound_filter_id).await?;
-        assert_eq!(new_upper_bound_logs, FilterChanges::Logs(upper_bound_logs));
+        assert_eq!(new_upper_bound_logs, FilterChanges::Hashes(vec![]));
```

Comment:
Nit: you can use

```rust
assert_matches!(_, FilterChanges::Hashes(hashes) if hashes.is_empty());
```

here and in other cases where empty changes are returned.